src/Microsoft.Health.Core/Features/HealthPublisher/HealthCheckMetricPublisher.cs

```diff
@@ -0,0 +1,53 @@
+// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System.Threading;
+using System.Threading.Tasks;
+using EnsureThat;
+using Microsoft.Extensions.Diagnostics.HealthChecks;
+using Microsoft.Extensions.Options;
+using Microsoft.Health.Core.Features.Health;
+using Microsoft.Health.Core.Features.Metric;
+
+namespace Microsoft.Health.Core.Features.HealthPublisher;
+internal class HealthCheckMetricPublisher : IHealthCheckPublisher
+{
+    private readonly IResourceHealthSignalProvider _resourceHealthMeter;
+    private readonly ResourceHealthDimensionOptions _resourceHealthDimensionOptions;
+
+    public HealthCheckMetricPublisher(IResourceHealthSignalProvider resourceHealthMeter, IOptions<ResourceHealthDimensionOptions> resourceHealthDimensionOptions)
+    {
+        EnsureArg.IsNotNull(resourceHealthDimensionOptions, nameof(resourceHealthDimensionOptions));
+
+        _resourceHealthDimensionOptions = EnsureArg.IsNotNull(resourceHealthDimensionOptions.Value, nameof(resourceHealthDimensionOptions.Value));
+        _resourceHealthMeter = EnsureArg.IsNotNull(resourceHealthMeter, nameof(resourceHealthMeter));
+
+        EnsureArg.IsNotNull(_resourceHealthDimensionOptions.ArmGeoLocation, nameof(_resourceHealthDimensionOptions.ArmGeoLocation));
+        EnsureArg.IsNotNull(_resourceHealthDimensionOptions.ResourceType, nameof(_resourceHealthDimensionOptions.ResourceType));
+        EnsureArg.IsNotNull(_resourceHealthDimensionOptions.ArmResourceId, nameof(_resourceHealthDimensionOptions.ArmResourceId));
+    }
+
+    public Task PublishAsync(HealthReport report, CancellationToken cancellationToken)
+    {
+        EnsureArg.IsNotNull(report, nameof(report));
+
+        switch (report.Status)
+        {
+            case HealthStatus.Healthy:
+                _resourceHealthMeter.EmitHealthMetric(HealthStatusReason.None, _resourceHealthDimensionOptions);
```

Comment:
So there is a 1:1 mapping for default reasons, but not for all reasons. The default reasons are there to answer "Hey, our HealthReport is x status but no Reason was defined in any of the health checks. What Reason should I publish?". We will pick a default Reason that provides a generic description of what is happening to the service (It's degraded or it's unavailable). 
But since we do sometimes specify a Reason (CMK) it's not 1:1.

& I don't want to put it in the health meter / signal provider so we can keep this basic logic the same between teams. FHIR will be using this as well. So we can keep the same severity order, same logic deciding which Reason to publish, and same experience when customers are viewing the Resource Health